### DISCUSSION TITLE
Custom Link method override error

### DISCUSSION THREAD

**Discussion by hugo-bernard** [clientIO/joint#2744] [Q&A]
### Introduction

Hello, i wanted to create a custom link class and potentially override some methods but a soon as my class is created i have errors on all methods of 'EventsMixin'. Here are the errors and class :
![image](https://github.com/user-attachments/assets/7d93145e-03a1-4fe3-8e16-d1c68a62b465)
```
export class CustomLink extends shapes.standard.Link {
    defaults() {
        return util.defaultsDeep({
            type: 'custom.Link',
            attrs: {
                line: {
                    stroke: 'blue',
                    strokeWidth: 2,
                    strokeDasharray: '5 5',
                    targetMarker: {
                        'type': 'path',
                        'd': 'M 10 -5 0 0 10 5 Z',
                        'stroke': 'blue',
                        'fill': 'blue'
                    }
                }
            }
        }, super.defaults);
    }
}
```

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS

**Comment by kumilingus** [comment]
How to reproduce it? If I copy the code into our _TypeScript_ test it works as expected.
